constr.mm

```diff
@@ -0,0 +1,139 @@
+$[ set.mm $]
+
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Constructible numbers
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  This section defines the set of constructible points as complex numbers which
+  can be drawn starting from two points (we take ` 0 ` and ` 1 ` ), and taking
+  intersections of circles and lines.
+
+  We initially define two sets:
+  <UL>
+  <LI>geometrically, by recursively adding the intersection points : this is
+    ` Constr ` .</LI>
+  <LI>algebraically, by using field extensions : this is ` ConstrAlg ` . </LI>
+  </UL>
+
+  We then proceed by showing that those two sets are equal.
+
+  This construction is useful for proving the impossibility of doubling the
+  cube ( ~ imp2cube ), and of angle trisection ( ~ imp3ang )
+$)
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Degree of a field extension
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    fldextdgirred.b $e |- B = ( Base ` F ) $.
+    fldextdgirred.0 $e |- .0. = ( 0g ` F ) $.
+    fldextdgirred.d $e |- D = ( deg1 ` F ) $.
+    fldextdgirred.o $e |- O = ( F evalSub1 E ) $.
+    fldextdgirred.m $e |- M = ( Poly1 ` F ) $.
+    fldextdgirred.k $e |- K = ( F |`s E ) $.
+    fldextdgirred.l $e |- L = ( F |`s ( F fldGen ( E u. { Z } ) ) ) $.
+    fldextdgirred.f $e |- ( ph -> F e. Field ) $.
+    fldextdgirred.e $e |- ( ph -> E e. ( SubDRing ` F ) ) $.
+    fldextdgirred.p $e |- ( ph -> P e. ( Base ` M ) ) $.
```

Comment:
```suggestion
    fldextdgirred.p $e |- ( ph -> P e. ( Poly1 ` ( F |`s E ) ) ) $.
```

```suggestion
    fldextdgirred.p $e |- ( ph -> P e. dom O ) $.
```

sorry, I just realized that technically P could be a polynomial with incorrect coefficients. As such ``( O ` P ) = (/)``, but since technically `.0. = (/)` is possible, we need to restrict `P` instead. Or we could have both the `evalSub` change and this change.